mmpose/models/backbones/rsn.py

```diff
@@ -0,0 +1,562 @@
+                        self.branch_channels,
+                        self.branch_channels,
+                        kernel_size=3,
+                        stride=1,
+                        padding=1,
+                        norm_cfg=self.norm_cfg,
+                        inplace=False))
+        self.conv_bn3 = ConvModule(
+            self.num_steps * self.branch_channels,
+            self.out_channels * self.expansion,
+            kernel_size=1,
+            stride=1,
+            padding=0,
+            act_cfg=None,
+            norm_cfg=self.norm_cfg,
+            inplace=False)
+        self.relu = nn.ReLU(inplace=False)
+
+    def forward(self, x):
+        """Forward function."""
+
+        def _inner_forward(x):
+            identity = x
+            x = self.conv_bn_relu1(x)
+            spx = torch.split(x, self.branch_channels, 1)
+            outputs = list()
+            outs = list()
+            for i in range(self.num_steps):
+                outputs_i = list()
+                outputs.append(outputs_i)
+                for j in range(i + 1):
+                    if j == 0:
+                        inputs = spx[i]
+                    else:
+                        inputs = outputs[i][j - 1]
+                    if i > j:
+                        inputs += outputs[i - 1][j]
+                    module_name = 'conv_bn_relu2_' + str(i + 1) + '_' + str(j +
+                                                                            1)
+                    module_i_j = getattr(self, module_name)
+                    outputs[i].append(module_i_j(inputs))
+
+                outs.append(outputs[i][i])
+            out = torch.cat(tuple(outs), 1)
+            out = self.conv_bn3(out)
+
+            if self.downsample is not None:
+                identity = self.downsample(identity)
+            out = out + identity
+
+            return out
+
+        if self.with_cp and x.requires_grad:
+            out = cp.checkpoint(_inner_forward, x)
+        else:
+            out = _inner_forward(x)
+
+        out = self.relu(out)
+
+        return out
+
+
+class Downsample_module(nn.Module):
+    """Downsample module for RSN.
+
+    Args:
+        block (nn.Module): Downsample block.
+        num_blocks (list): Number of blocks in each downsample unit.
+        num_units (int): Numbers of downsample units. Default: 4
+        has_skip (bool): Have skip connections from prior upsample
+            module or not. Default:False
+        num_steps (int): Number of steps in a block. Default:4
+        norm_cfg (dict): dictionary to construct and config norm layer.
+            Default: dict(type='BN')
+    """
+
+    def __init__(self,
+                 block,
+                 num_blocks,
+                 num_steps=4,
+                 num_units=4,
+                 has_skip=False,
+                 norm_cfg=dict(type='BN')):
+        super().__init__()
+        self.has_skip = has_skip
+        self.in_channels = 64
```

Comment:
Can we make `in_channels` an input variable?

Reply:
Yes. in_channels represents the number of channels of the feature from ResNet_top.